### PR TITLE
add a Dockerfile

### PR DESCRIPTION
**Pull request by unclejack**
This PR adds a Dockerfile to build the website.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM ubuntu:16.04
+RUN apt-get update && apt-get install -y git-core openssh-client curl wget \
+ build-essential openssl libreadline6 libreadline6-dev curl zlib1g zlib1g-dev \
+ libssl-dev libyaml-dev libsqlite3-dev sqlite3 libxml2-dev libxslt-dev \
+ autoconf libc6-dev ncurses-dev automake libtool bison pkg-config \
+ gawk libgmp-dev libgdbm-dev libffi-dev
+RUN curl -L https://get.rvm.io | bash -s stable
+RUN /bin/bash -l -c "rvm requirements"
```
Comment (dseevr) [inline Dockerfile]: If you add this:

```sh
echo "gem: --no-document" >~/.gemrc
```

The gem/bundle commands will skip installing docs which will speed them up by 2-3x

Comment (unclejack) [inline Dockerfile]: @dseevr That's a really good idea! I've updated the PR.

**Comment by dseevr** [comment]
@unclejack the Dockerfile builds fine for me now

Could we update the Makefile targets to leverage this new container image, too?

**Comment by unclejack** [comment]
@dseevr Do you have any preferences regarding the location of the built site?

**Comment by dseevr** [comment]
@unclejack nope

**Comment by stacykor** [comment]
Hi @unclejack @dseevr , are there any more changes planned to the Dockerfile?

**Comment by dseevr** [comment]
To the `Dockerfile`, no, but the `Makefile` needs to be updated to use the Docker image rather than expecting Ruby/Bundler/gems to be installed on the host OS

**Comment by dseevr** [comment]
This can be merged and the `Makefile` changes can be done in a future PR

**Pull request merged**